www/js/metrics/metricsHelper.ts

```diff
@@ -0,0 +1,212 @@
+import { DateTime } from "luxon";
+import { formatForDisplay } from "../config/useImperialConfig";
+import { DayOfMetricData } from "./metricsTypes";
+import moment from 'moment';
+
+export function getUniqueLabelsForDays(metricDataDays: DayOfMetricData[]) {
+  const uniqueLabels: string[] = [];
+  metricDataDays.forEach(e => {
+    Object.keys(e).forEach(k => {
+      if (k.startsWith('label_')) {
+        const label = k.substring(6); // remove 'label_' prefix leaving just the mode label
+        if (!uniqueLabels.includes(label)) uniqueLabels.push(label);
+      }
+    });
+  });
+  return uniqueLabels;
+}
+
+export const getLabelsForDay = (metricDataDay: DayOfMetricData) => (
+  Object.keys(metricDataDay).reduce((acc, k) => {
+    if (k.startsWith('label_')) {
+      acc.push(k.substring(6)); // remove 'label_' prefix leaving just the mode label
+    }
+    return acc;
+  }, [] as string[])
+);
+
+export const secondsToMinutes = (seconds: number) =>
+  formatForDisplay(seconds / 60);
+
+export const secondsToHours = (seconds: number) =>
+  formatForDisplay(seconds / 3600);
+
+// segments metricsDays into weeks, with the most recent week first
+export function segmentDaysByWeeks (days: DayOfMetricData[], nWeeks?: number) {
+  const weeks: DayOfMetricData[][] = [];
+  for (let i = days?.length - 1; i >= 0; i -= 7) {
+    weeks.push(days.slice(Math.max(i - 6, 0), i + 1));
+  }
+  if (nWeeks) return weeks.slice(0, nWeeks);
+  return weeks;
+};
+
+export function formatDate(day: DayOfMetricData) {
```

Comment:
isn't this in/overlapping with diaryHelper?

Reply:
Kind of. This is all using Luxon (ie `DateTime`) but `diaryHelper` is still using Moment in a few places, which should be converted eventually

And they're not exactly the same date format because the day of week "Mon" or "Monday" is used on the label tab while just numeric dates are used on the dashboard.

But still at least some unification can be done, I think.